### DISCUSSION TITLE
Where is support for hljs lang-restApi coming from?

### DISCUSSION THREAD

**Discussion by mikernet** [dotnet/docfx#8187] [Q&A] [answered]
We have a custom v2 theme that is not doing code highlighting for REST API code blocks that have `class="lang-restApi hljs"` and I can't for the life of me find where support for lang-restApi is coming from in the default theme nor can I find anything online about a highlight.js REST API language I can plug in. When using my theme I get:

![image](https://user-images.githubusercontent.com/1700111/192080306-e9bb3d99-eb13-4a50-871c-b5f989182ca4.png)

Where is that language support coming from so I can include it in my theme?

**Comment by paulushub** [comment, marked as answer]
Try HTTP, with REST API it is either HTTP or JSON.

On the demo site https://highlightjs.org/static/demo/, select the `web` or `protocols` in the **Language Categories**.

**Reply by mikernet** [reply]
That's the html code that docfx outputs and it somehow works on the default theme. I'm trying to figure out how to make it work on my theme as well the same way it does in the default theme, without having to change that output.

**Reply by mikernet** [reply]
I looked everywhere and searched in all the default theme files but can't find anything that would wire up hljs to support `lang-restApi` but I'm obviously missing something, it's gotta be there somewhere.

**Reply by paulushub** [reply]
default/partials/rest.child.tmpl.partial(26,26)  
```
<pre>
    <code class="lang-restApi hljs">{{operation}} {{path}}</code>
</pre>
```

**Reply by KalleOlaviNiemitalo** [reply]
Is there a site demonstrating that highlighting of lang-restApi even works with the default theme/template?

**Reply by KalleOlaviNiemitalo** [reply]
<https://learn.microsoft.com/en-us/rest/api/advisor/configurations/create-in-resource-group> uses `<code class="lang-http" dir="ltr">` rather than `lang-restApi`. I guess that was built with DocFX v3.

**Reply by paulushub** [reply]
> Is there a site demonstrating that highlighting of lang-restApi even works with the default theme/template?

See this sample: http://docascode.github.io/docfx-seed/restapi/petstore.html
Select the `REST API`, you can view the source codes.

**Reply by mikernet** [reply]
@paulushub I know that's the part that makes it output `class="lang-restApi hljs"`, the issue is how is that getting highlighting instead of the error in the OP?

**Reply by paulushub** [reply]
> the issue is how is that getting highlighting instead of the error in the OP?

I do not see any syntax highlighting with the sample site I posted. If you build that sample yourself, and edit the `_site/restapi/petstore.html` to replace `lang-restApi` with `lang-http` you will see the difference.

**Reply by KalleOlaviNiemitalo** [reply]
<http://docascode.github.io/docfx-seed/restapi/petstore.html#petstore_swagger_io_v2_Swagger_Petstore_1_0_0_deletePet> includes `<code class="lang-restApi hljs">DELETE /pet/{petId}</code>` but I don't see any highlighting for that.

The Pet Store page loaded a docfx.vendor.json file that mentions "highlight.js v9.12.0". The _"Could not find the language"_ message was added with <https://github.com/highlightjs/highlight.js/pull/2236> in highlight.js 9.17.0, which would explain why no warning is logged.

**Reply by mikernet** [reply]
Oh, I see...I have no idea what the issue submitted to my theme repo was talking about then, haha. I think we can mark this as resolved for now, Sorry for the confusion on my part.
